Docs: Iterate on caching APIs pages, add `cacheLife` function page (#72081)
vercel/next.js: docs/02-app/02-api-reference/04-functions/cacheLife.mdx

---
title: cacheLife
description: Learn how to use the cacheLife function to set the cache expiration time for a cached function or component.
version: canary
related:
  title: Related
  description: View related API references.
  links:
    - app/api-reference/next-config-js/dynamicIO
    - app/api-reference/directives/use-cache
    - app/api-reference/functions/revalidateTag
    - app/api-reference/functions/cacheTag
---

> The `cacheLife` function is used to set tge cache lifetime of a function or component. It should be used alongside the `use cache` directive, and within the scope of the function or component.

Reviewer: ```suggestion
The `cacheLife` function is used to set the cache lifetime of a function or component. It should be used alongside the [`use cache`](/docs/canary/app/api-reference/directives/use-cache) directive, and within the scope of the function or component.
```

## Usage

> To use `cacheLife`, enable the [`dynamicIO` flag](/docs/app/api-reference/next-config-js/dynamicIO) in your `next.config.js` file and import `cacheLife` from `next/cache`.

Reviewer: Can remove the imort suggestion, as you go on to state this with an example under setting the flag
```suggestion
To use `cacheLife`, enable the [`dynamicIO` flag](/docs/app/api-reference/next-config-js/dynamicIO) in your `next.config.js` file:
```

```ts filename="next.config.ts" switcher
import type { NextConfig } from 'next'

const nextConfig: NextConfig = {
  experimental: {
    dynamicIO: true,
  },
}

export default nextConfig
```

```js filename="next.config.js" switcher
const nextConfig = {
  experimental: {
    dynamicIO: true,
  },
}

export default nextConfig
```

> Then, invoke The `cacheLife` function within the scope of the function or component:

Reviewer: ```suggestion
Then, import and invoke The `cacheLife` function within the scope of the function or component:
```

```tsx filename="app/page.tsx" highlight={5} switcher
'use cache'
import { unstable_cacheLife as cacheLife } from 'next/cache'

export default async function Page() {
  cacheLife('hours')
  return <div>Page</div>
}
```

```tsx filename="app/page.tsx" highlight={5} switcher
'use cache'
import { unstable_cacheLife as cacheLife } from 'next/cache'

export default async function Page() {
  cacheLife('hours')
  return <div>Page</div>
}
```

> ## How cache revalidation works

Author: Hm, this should probably go into the deep dive as well

Reviewer: Yeah, this doesnt feel like ref doc material. Good shout

'use cache'
import { unstable_cacheLife as cacheLife } from 'next/cache'

export default async function Page() {
  cacheLife('biweekly')
  return <div>Page</div>
}
```

### Overriding the default cache profiles

While the default cache profiles provide a useful way to think about how fresh or stale any given part of cacheable output can be, you may prefer different named profiles to better align with your applications caching strategies.

You can override the default named cache profiles by creating a new configuration with the same name as the defaults.

The example below shows how to override the default “days” cache profile:

```ts filename="next.config.ts"
const nextConfig = {
  experimental: {
    dynamicIO: true,
    cacheLife: {
      days: {
        stale: 3600, // 1 hour
        revalidate: 900, // 15 minutes
        expire: 86400, // 1 day
      },
    },
  },
}

module.exports = nextConfig
```

### Defining cache profiles inline

For specific use cases, you can set a custom cache profile by passing an object to the `cacheLife` function:

```tsx filename="app/page.tsx" highlight={5,6,7}
'use cache'
import { unstable_cacheLife as cacheLife } from 'next/cache'

> cacheLife({

Reviewer: Given that you cannot actually call the function outside the scope of a function, should we put this example within a function to avoid any unintended misunderstandings?

Author: Good catch, will update!
